### PR TITLE
Add support for GUID binding to relative UNIX-style file paths for asset tracking.

### PR DESCRIPTION
**Pull request by RubyNova**

### CODE REVIEW

File: include/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.h [hunk 1]
```diff
@@ -15,8 +15,12 @@ namespace NovelRT::ResourceManagement::Desktop
     private:
         LoggingService _logger;
 
+    protected:
+        void WriteAssetDatabaseFile() final;
+        void LoadAssetDatabaseFile() override;
+
     public:
-        [[nodiscard]] TextureMetadata LoadTextureFromFile(std::filesystem::path filePath) final;
+        [[nodiscard]] TextureMetadata LoadTexture(std::filesystem::path filePath) final;
```
Comment (capnkenny) [inline include/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.h:23]: Is this to be an example for future file loading for resources? More of a curiosity question

File: src/NovelRT/ResourceManagement/ResourceLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                guid = gen();
+            }
+
+            guidPathMap.emplace(guid, filePath);
+            pathGuidMap.emplace(filePath, guid);
+        }
+
+        WriteAssetDatabaseFile();
+    }
+}
```
Comment (capnkenny) [inline src/NovelRT/ResourceManagement/ResourceLoader.cpp:44]: Nit: newline

File: include/NovelRT/Utilities/Misc.h [hunk 3]
```diff
@@ -60,6 +60,25 @@ namespace NovelRT::Utilities
...
+            while (end != std::string::npos)
+            {
+                returnVec.emplace_back(targetString.substr(start, end - start));
+                start = end + delimiter.size();
+                end = targetString.find(delimiter, start);
+            }
+
+            returnVec.emplace_back(targetString.substr(start)); // THIS LINE ANGERS ME
+
+            return returnVec;
```
Comment (capnkenny) [inline include/NovelRT/Utilities/Misc.h:80]: is this safe just because it's from a static method? (Meaning that we don't lose it on the stack)

**Review by capnkenny — CHANGES_REQUESTED** [review]
Generally looks okay - what will be generating the Asset file, though? Is it to be generated by the Engine? Or is it to be hand-made first, and then eventually loaded?

File: include/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.h (hunk 1 above)
Comment (RubyNova) [reply include/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.h:23]: This was more to bring it in line with the other resource loading methods, it looked weird. :D

Comment (capnkenny) [reply include/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.h:23]: No worries - Audio is currently considered a Resource but the loading method is not here, so this was more of a "do we have a future TODO" q

File: include/NovelRT/Utilities/Misc.h (hunk 3 above)
Comment (capnkenny) [reply include/NovelRT/Utilities/Misc.h:80]: yes

Comment (RubyNova) [reply include/NovelRT/Utilities/Misc.h:80]: why wouldn't it be safe? a vector is not a span that's having its underlying memory being yanked. Aren't we already doing this in other utility methods?

Comment (capnkenny) [reply include/NovelRT/Utilities/Misc.h:80]: I just forgot that this is copying value and not returning a ref 🙃 sorry about that!

**Review by capnkenny — APPROVED** [review]

**Pull request merged**